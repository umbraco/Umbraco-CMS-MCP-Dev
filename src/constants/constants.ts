export const BLANK_UUID = "00000000-0000-0000-0000-000000000000";
export const ROOT_DOCUMENT_TYPE_ID = "a95360e8-ff04-40b1-8f46-7aa4b5983096";
export const CONTENT_DOCUMENT_TYPE_ID = "b871f83c-2395-4894-be0f-5422c1a71e48";
export const Default_Memeber_TYPE_ID = "d59be02f-1df9-4228-aa1e-01917d806cda";
export const TextString_DATA_TYPE_ID = "0cc0eba1-9960-42c9-bf9b-60e150b429ae";
export const IMAGE_MEDIA_TYPE_ID = "cc07b313-0843-4aa8-bbda-871c8da728c8";
export const FOLDER_MEDIA_TYPE_ID = "f38bd2d7-65d0-48e6-95dc-87ce06ec2d3d";
export const EXAMPLE_IMAGE_PATH =
  "/src/umb-management-api/tools/temporary-file/__tests__/helpers/example.jpg";
