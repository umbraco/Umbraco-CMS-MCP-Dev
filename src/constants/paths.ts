import { join } from 'path';

// Use absolute path from current working directory
export const RESOURCES_DIR = join(process.cwd(), 'build', 'resources').replace(/\\/g, '/');