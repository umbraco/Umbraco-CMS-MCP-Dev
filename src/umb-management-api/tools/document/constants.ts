export const UmbracoDocumentPermissions = {
    Create: "Umb.Document.Create",
    Read: "Umb.Document.Read",
    Update: "Umb.Document.Update",
    Delete: "Umb.Document.Delete",
    Publish: "Umb.Document.Publish",
    Unpublish: "Umb.Document.Unpublish",
    Move: "Umb.Document.Move",
    Sort: "Umb.Document.Sort",
    Duplicate: "Umb.Document.Duplicate",
    PublicAccess: "Umb.Document.PublicAccess",
    CultureAndHostnames: "Umb.Document.CultureAndHostnames",
    CreateBlueprint: "Umb.Document.CreateBlueprint",
}