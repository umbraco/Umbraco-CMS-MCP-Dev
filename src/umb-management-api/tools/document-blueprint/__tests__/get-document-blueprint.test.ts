import GetDocumentBlueprintTool from "../get/get-blueprint.js";
import { DocumentBlueprintBuilder } from "./helpers/document-blueprint-builder.js";
import { DocumentBlueprintTestHelper } from "./helpers/document-blueprint-test-helper.js";
import { jest } from "@jest/globals";
import { BLANK_UUID } from "@/constants/constants.js";

describe("get-document-blueprint", () => {
  const TEST_BLUEPRINT_NAME = "_Test Blueprint Get";
  let originalConsoleError: typeof console.error;

  beforeEach(() => {
    originalConsoleError = console.error;
    console.error = jest.fn();
  });

  afterEach(async () => {
    console.error = originalConsoleError;
    await DocumentBlueprintTestHelper.cleanup(TEST_BLUEPRINT_NAME);
  });

  it("should get a document blueprint by id", async () => {
    // Create a blueprint to get
    const builder = await new DocumentBlueprintBuilder(
      TEST_BLUEPRINT_NAME
    ).create();

    // Get the blueprint
    const result = await GetDocumentBlueprintTool().handler(
      {
        id: builder.getId(),
      },
      { signal: new AbortController().signal }
    );

    // Normalize dates and IDs in the response
    const normalizedResult = {
      ...result,
      content: result.content.map((content) => {
        const parsed = JSON.parse(content.text as string);
        return {
          ...content,
          text: JSON.stringify({
            ...parsed,
            id: BLANK_UUID,
            documentType: {
              ...parsed.documentType,
              id: BLANK_UUID,
            },
            variants: parsed.variants.map((variant: any) => ({
              ...variant,
              createDate: "2000-01-01T00:00:00.000Z",
              updateDate: "2000-01-01T00:00:00.000Z",
            })),
          }),
        };
      }),
    };

    // Verify the handler response using snapshot
    expect(normalizedResult).toMatchSnapshot();
  });

  it("should handle non-existent document blueprint", async () => {
    const result = await GetDocumentBlueprintTool().handler(
      {
        id: BLANK_UUID,
      },
      { signal: new AbortController().signal }
    );

    // Verify the error response using snapshot
    expect(result).toMatchSnapshot();
  });
});
