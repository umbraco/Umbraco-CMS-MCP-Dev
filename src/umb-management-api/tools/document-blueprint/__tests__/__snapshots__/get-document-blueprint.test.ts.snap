// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`get-document-blueprint should get a document blueprint by id 1`] = `
{
  "content": [
    {
      "text": "{"documentType":{"id":"00000000-0000-0000-0000-000000000000","icon":"icon-home color-blue","collection":null},"id":"00000000-0000-0000-0000-000000000000","values":[],"variants":[{"state":"Draft","publishDate":null,"scheduledPublishDate":null,"scheduledUnpublishDate":null,"createDate":"2000-01-01T00:00:00.000Z","updateDate":"2000-01-01T00:00:00.000Z","culture":null,"segment":null,"name":"_Test Blueprint Get"}]}",
      "type": "text",
    },
  ],
}
`;

exports[`get-document-blueprint should handle non-existent document blueprint 1`] = `
{
  "content": [
    {
      "text": "Error using get-document-blueprint:
{
  "message": "Request failed with status code 404",
  "response": {
    "type": "Error",
    "title": "The document blueprint could not be found",
    "status": 404,
    "operationStatus": "NotFound"
  }
}",
      "type": "text",
    },
  ],
}
`;
